addons/fleet/security/fleet_security.xml

```diff
@@ -20,66 +20,48 @@
         <record id="base.default_user" model="res.users">
             <field name="groups_id" eval="[(4,ref('fleet.fleet_group_manager'))]"/>
         </record>
-        <record id="fleet_rule_contract_visibility_user" model="ir.rule">
```

Comment:
Those rules are noupdate="1"

You need to force the update in a migration script by using ``util.update_record_from_xml``